aspnetcore/security/authorization/iauthorizationpolicyprovider.md

```diff
@@ -0,0 +1,152 @@
+---
+title: Custom Authorization Policy Providers in ASP.NET Core
+author: mjrousos
+description: Learn how to use a custom IAuthorizationPolicyProvider in an ASP.NET Core app to dynamically generate authorization policies.
+manager: wpickett
+ms.author: riande
+ms.custom: mvc
+ms.date: 05/02/2018
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: article
+uid: security/authorization/iauthorizationpolicyprovider
+---
+# Custom Authorization Policy Providers using IAuthorizationPolicyProvider in ASP.NET Core 
+
+By [Mike Rousos](https://github.com/mjrousos)
+
+Typically when using [policy-based authorization](xref:security/authorization/policies), policies are registered by calling `AuthorizationOptions.AddPolicy` as part of authorization service configuration. In some scenarios, it may not be possible (or desirable) to register all authorization policies in this way. In those cases, you can use a custom `IAuthorizationPolicyProvider` to control how authorization policies are supplied.
+
+Examples of scenarios where a custom [IAuthorizationPolicyProvider](/dotnet/api/microsoft.aspnetcore.authorization.iauthorizationpolicyprovider) may be useful include:
+
+* Creating policies at runtime based on information in an external data source (like a database) or determining authorization requirements dynamically through another mechanism.
+* Using a large range of policies (for different room numbers or ages, for example), so it doesn’t make sense to add each individual authorization policy with an `AuthorizationOptions.AddPolicy` call.
```

Comment:
Add "* Using an external service to provide policy evaluation" (or something of that ilk)

Reply:
Good addition. I've updated the doc.